ethereum/contracts/relayer/deliveryProvider/DeliveryProvider.sol

```diff
@@ -113,28 +114,37 @@ contract DeliveryProvider is DeliveryProviderGovernance, IDeliveryProvider {
      */
 
     //Returns the delivery overhead fee required to deliver a message to the target chain, denominated in this chain's wei.
-    function quoteDeliveryOverhead(uint16 targetChain) public view returns (Wei nativePriceQuote) {
-        Gas overhead = deliverGasOverhead(targetChain);
-        Wei targetFees = overhead.toWei(gasPrice(targetChain));
-        Wei result = assetConversion(targetChain, targetFees, chainId());
-        require(result.unwrap() <= type(uint128).max, "Overflow");
-        return result;
+    function quoteDeliveryOverhead(uint16 targetChain) public view returns (LocalNative nativePriceQuote) {
+        nativePriceQuote = quoteGasCost(targetChain, deliverGasOverhead(targetChain));
+        require(nativePriceQuote.unwrap() <= type(uint128).max, "Overflow");
+    }
+
+    //Returns the price of purchasing gasAmount units of gas on the target chain, denominated in this chain's wei.
+    function quoteGasCost(uint16 targetChain, Gas gasAmount) public view returns (LocalNative totalCost) {
+        Wei gasCostInSourceChainCurrency =
+            assetConversion(targetChain, gasAmount.toWei(gasPrice(targetChain)), chainId());
+        totalCost = LocalNative.wrap(gasCostInSourceChainCurrency.unwrap());
     }
 
-    //Returns the price of purchasing 1 unit of gas on the target chain, denominated in this chain's wei.
-    function quoteGasPrice(uint16 targetChain) public view returns (GasPrice) {
-        Wei gasPriceInSourceChainCurrency =
-            assetConversion(targetChain, gasPrice(targetChain).priceAsWei(), chainId());
-        require(gasPriceInSourceChainCurrency.unwrap() <= type(uint88).max, "Overflow");
-        return GasPrice.wrap(uint88(gasPriceInSourceChainCurrency.unwrap()));
+    function quoteGasPrice(uint16 targetChain) public view returns (GasPrice price) {
+        price = GasPrice.wrap(quoteGasCost(targetChain, Gas.wrap(1)).unwrap());
+        require(price.unwrap() <= type(uint88).max, "Overflow");
     }
 
+    error PriceIsZero(uint16 chain);
```

Comment:
Nit: move this to interface, or top of file

Reply:
done